### PR TITLE
Add cases list page to trs support UI

### PR DESCRIPTION
**Pull request by hortha**
### Context

We’re adding support for resolving Change of name/DOB cases via the TRS Support Console. This page is the list of active cases.

### Changes proposed in this pull request

Add a new page at /cases with title ‘Open cases’.

Add a table with columns ‘Case reference’, ‘Name', ‘Case type’ and ‘Date’. Populate this table with Active incidententities where the subject is either ‘Change of Name’ or ‘Change of Date of Birth’. ‘Case reference’ is theticketnumberattribute, ‘Name’ is the current first name + last name of the relatedcontact(use the stated first/name names if set), ‘Case type’ is the subject title and ‘Date’ is the cases’ created date. Rows should be sorted oldest to newest. The ‘Case reference’ cell should be a link to/cases/{id} (a new page). The ‘Case reference', ‘Case type’ and ‘Date’ columns should have fixed width.

If there are no cases show a row with the content ‘No cases’.

No paging is required for now.

Add a ‘Cases’ option to the nav bar linking to this new page.

### Guidance to review

Changes to UI + Dataverse + tests (including test helper classes).

### Checklist

-   [x] Attach to Trello card
-   [x] Rebased master
-   [ ] Cleaned commit history
-   [x] Tested by running locally

### CODE REVIEW

**Comment by hortha** [comment]
[Add cases list page to TRS Support UI](https://trello.com/c/QFLA4Gae/1176-add-cases-list-page-to-trs-support-ui)

File: TeachingRecordSystem/TeachingRecordSystem.sln [hunk 1]
```diff
@@ -31,7 +31,7 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "TeachingRecordSystem.Core.T
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "TeachingRecordSystem.SupportUi.EndToEndTests", "tests\TeachingRecordSystem.SupportUi.EndToEndTests\TeachingRecordSystem.SupportUi.EndToEndTests.csproj", "{EB46905A-F7B4-4BDC-B376-6091264D6303}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "TeachingRecordSystem.SupportUi.Tests", "tests\TeachingRecordSystem.SupportUi.Tests\TeachingRecordSystem.SupportUi.Tests.csproj", "{407EB5A5-8B4F-4F7D-80BA-E39AB6AD013C}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "TeachingRecordSystem.SupportUi.Tests", "tests\TeachingRecordSystem.SupportUi.Tests\TeachingRecordSystem.SupportUi.Tests.csproj", "{407EB5A5-8B4F-4F7D-80BA-E39AB6AD013C}"
```
Comment (gunndabad) [inline TeachingRecordSystem/TeachingRecordSystem.sln:34]: Nit: can we undo this change?

Comment (hortha) [reply TeachingRecordSystem/TeachingRecordSystem.sln:34]: I keep doing it but not sure why it keeps re-applying it!

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Cases/Index.cshtml.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            CreatedOn = incident.CreatedOn!.Value
+        };
+    }
+
+    public record CaseInfo
+    {
+        public required string CaseReference { get; set; }
+        public required string Customer { get; set; }
+        public required string CaseType { get; set; }
+        public required DateTime CreatedOn { get; set; }
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Cases/Index.cshtml.cs:47]:
```suggestion
        public required string CaseReference { get; init; }
        public required string Customer { get; init; }
        public required string CaseType { get; init; }
        public required DateTime CreatedOn { get; init; }
```

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Program.cs [hunk 3]
```diff
@@ -137,6 +139,12 @@
     };
 
     builder.Services.AddTransient<ServiceClient>(_ => serviceClient.Clone());
+    builder.Services
+        .AddSingleton<ITrnGenerationApiClient, TrnGenerationApiClient>() // Purely needed to DI into DataverseAdapter
+        .AddTransient<IOrganizationServiceAsync>(_ => serviceClient.Clone())
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Program.cs:144]:
```suggestion
        .AddTransient<IOrganizationServiceAsync>(sp => sp.GetRequiredService<ServiceClient>())
```

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Program.cs [hunk 4]
```diff
@@ -137,6 +139,12 @@
     };
 
     builder.Services.AddTransient<ServiceClient>(_ => serviceClient.Clone());
+    builder.Services
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Program.cs:142]: Can we either chain all of these `Add*` calls together or have separate statements for each?

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/TrsLinkGenerator.cs [hunk 5]
```diff
@@ -15,6 +15,11 @@ public TrsLinkGenerator(LinkGenerator linkGenerator)
 
     public string SignedOut() => GetRequiredPathByPage("/SignedOut");
 
+    public string Cases() => GetRequiredPathByPage("/Cases/Index");
+
+    // Temporarily not making this required as the page hasn't been created yet.
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/TrsLinkGenerator.cs:20]: Does this comment still apply?

File: TeachingRecordSystem/TeachingRecordSystem.sln (hunk 1 above)
Comment (hortha) [reply TeachingRecordSystem/TeachingRecordSystem.sln]: done

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Cases/Index.cshtml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+            <tr class="govuk-table__row" data-testid="no-cases">
+                 <td class="govuk-table__cell" colspan="4">No cases</td>
+            </tr>
+        }
+        else
+        {
+            @foreach (var caseInfo in Model.Cases!)
+            {
+                <tr class="govuk-table__row" data-testid="case-@caseInfo.CaseReference">
+                    <td class="govuk-table__cell" data-testid="case-reference-@caseInfo.CaseReference"><a href="@LinkGenerator.EditCase(caseInfo.CaseReference)">@caseInfo.CaseReference</a></td>
```
[the remark this answers is not in the document]
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Cases/Index.cshtml]: done

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Cases/Index.cshtml.cs (hunk 2 above)
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Cases/Index.cshtml.cs]: done

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Program.cs (hunk 3 above)
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Program.cs]: done

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/TrsLinkGenerator.cs (hunk 5 above)
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/TrsLinkGenerator.cs]: nope
